### DISCUSSION TITLE
Payment receipt not receiving

### DISCUSSION THREAD

**Discussion by ChathuraCodimite** [hyochan/react-native-iap#2248] [Q&A]
Here is my code and problem is not receiving the receipt after payment completed
Any help?

const _initIAP = useCallback(async (): Promise<void> => {
    clearProductsIOS();

    try {
      const result = await initConnection();
      await flushFailedPurchasesCachedAsPendingAndroid();

      if (result === false) {
        Alert.alert("couldn't get in-app-purchase information");
        return;
      }

    } catch (error) {
      console.debug('initConnection');
      console.error(error);
      Alert.alert('fail to get in-app-purchase information');
    }

    purchaseUpdateSubscription = purchaseUpdatedListener(
      (purchase: SubscriptionPurchase | ProductPurchase) => {
        console.log('purchaseUpdatedListener : ', purchase);
        console.log('receipt : ', purchase.transactionReceipt);
        const receipt = purchase.transactionReceipt;
        if (receipt) {
          finishTransaction({purchase, isConsumable: true})
        }
      },
    );

    purchaseErrorSubscription = purchaseErrorListener(
      (error: PurchaseError) => {
        console.warn('purchaseErrorListener', error);
      },
    );

    handleGetSubscriptions()
  }, []);

**Comment by andresesfm** [comment]
Not sure what "not receiving the receipt" means. Is it getting undefined on the listener or the listener not being called

**Comment by baesumin** [comment]
+1  listener not being called on ios somtimes.
When the callback is not called, it seems that there is no iOS purchase receipt. In this case, how can the client receive the purchase data and pass it to the server?
